src/WtsApi32.Desktop/WtsApi32.cs

```diff
@@ -0,0 +1,112 @@
+        /// server, or <see cref="WTS_CURRENT_SERVER_HANDLE"/> to use the RD Session Host server that hosts your application.</param>
+        /// <param name="reserved">This parameter is reserved. It must be zero.</param>
+        /// <param name="version">The version of the enumeration request. This parameter must be 1.</param>
+        /// <param name="ppSessionInfo">A pointer to an array of <see cref="WTS_SESSION_INFO"/> structures that represent the retrieved
+        /// sessions. To free the returned buffer, call the <see cref="WTSFreeMemory"/> function.</param>
+        /// <param name="pSessionInfoCount">A pointer to the number of WTS_SESSION_INFO structures returned in the ppSessionInfo parameter.</param>
+        /// <returns>Returns zero if this function fails. If this function succeeds, a nonzero value is returned.</returns>
+        [DllImport(nameof(WtsApi32), SetLastError = true, CharSet = CharSet.Auto)]
+        public static extern bool WTSEnumerateSessions(
+            IntPtr hServer,
+            [MarshalAs(UnmanagedType.U4)] int reserved,
+            [MarshalAs(UnmanagedType.U4)] int version,
+            ref IntPtr ppSessionInfo,
+            [MarshalAs(UnmanagedType.U4)] ref int pSessionInfoCount);
+
+        /// <summary>
+        /// Retrieves a list of sessions on a Remote Desktop Session Host (RD Session Host) server.
+        /// </summary>
+        /// <param name="hServer">A handle to the RD Session Host server.
+        /// You can use the <see cref="WTSOpenServer"/> or <see cref="WTSOpenServerEx"/> functions to retrieve a handle to a specific
+        /// server, or <see cref="WTS_CURRENT_SERVER_HANDLE"/> to use the RD Session Host server that hosts your application.</param>
+        /// <param name="reserved">This parameter is reserved. It must be zero.</param>
+        /// <param name="version">The version of the enumeration request. This parameter must be 1.</param>
+        /// <param name="ppSessionInfo">A pointer to <see cref="IEnumerable&lt;WTS_SESSION_INFO&gt;"/> structures that represent the retrieved
+        /// sessions. Note, that returned object doesn't know overall count of sessions, and always return true for MoveNext, use it in pair
+        /// with pSessionInfoCount parameter</param>
+        /// <param name="pSessionInfoCount">A pointer to the number of WTS_SESSION_INFO structures returned in the ppSessionInfo parameter.</param>
+        /// <returns>Returns zero if this function fails. If this function succeeds, a nonzero value is returned.</returns>
+        [DllImport(nameof(WtsApi32), SetLastError = true, CharSet = CharSet.Auto)]
+        public static extern bool WTSEnumerateSessions(
+            IntPtr hServer,
+            [MarshalAs(UnmanagedType.U4)] int reserved,
+            [MarshalAs(UnmanagedType.U4)] int version,
+            ref WtsSafeSessionInfoGuard ppSessionInfo,
```

Comment:
The methods we expose are as close as possible to the native API, we use `SafeHandle` when they are adapted but they aren't here.

Your method can't work as an `extern` / `DllImport` as the marshaler wouldn't know that it must provide the `pSessionInfoCount` argument or from where to take it.

But as said before such a method is perfect for a helper. We already have quite used exactly for this purpose: 

* https://github.com/AArnott/pinvoke/blob/master/src/Kernel32.Shared/Kernel32.Helpers.cs#L171
* https://github.com/AArnott/pinvoke/blob/64b5b888d031ab6b54e72f4646693250883d55f0/src/AdvApi32.Desktop/AdvApi32.Helpers.cs#L29